sdk/core/azure-core/src/main/java/com/azure/core/util/ClientOptions.java

```diff
@@ -24,6 +26,40 @@
 
     private String applicationId;
 
+    public static ClientOptions fromConfiguration(Configuration configuration) {
+        if (configuration == Configuration.NONE) {
+            return new ClientOptions();
+        }
+
+        Configuration clientConfiguration = (configuration == null)
+            ? Configuration.getGlobalConfiguration()
+            : configuration;
+
+        ClientOptions options = new ClientOptions();
+        loadClientOptions("client", options, clientConfiguration);
+
+        return options;
+    }
+
+    static void loadClientOptions(String prefix, ClientOptions clientOptions, Configuration configuration) {
+
+        // Only use system proxies when the prerequisite property is 'true'.
+        String applicationId = configuration.get(prefix + ".application-id");
```

Comment:
Should we introduce a convenience API on `Configuration` of `get(String prefix, String key)`?